.github/workflows/cmake.yml

```diff
@@ -0,0 +1,45 @@
+name: CMake
+
+on:
+  push:
+    branches:
+      - master
```

Comment:
Wouldn't you want to run on non-default branches when pushed to the repo, as well? ISTM this is never going to catch problems before they're merged to master.